keep-ui/app/incidents/[id]/incident-workflow-sidebar.tsx

```diff
@@ -0,0 +1,176 @@
+import { Fragment, useEffect } from "react";
+import { Dialog, Transition } from "@headlessui/react";
+import { Text, Button, TextInput, Badge, Title, Card } from "@tremor/react";
+import { IoMdClose } from "react-icons/io";
+import { WorkflowExecution } from "app/workflows/builder/types";
+import { formatDistanceToNowStrict } from "date-fns";
+import {
+  getIcon,
+  getTriggerIcon,
+  extractTriggerValue,
+} from "app/workflows/[workflow_id]/workflow-execution-table";
+import { useWorkflowExecution } from "utils/hooks/useWorkflowExecutions";
+
+interface IncidentWorkflowSidebarProps {
+  isOpen: boolean;
+  toggle: VoidFunction;
+  selectedExecution: WorkflowExecution | null;
+}
+
+const IncidentWorkflowSidebar: React.FC<IncidentWorkflowSidebarProps> = ({
+  isOpen,
+  toggle,
+  selectedExecution,
+}) => {
+  const { data: workflowExecutionData } = useWorkflowExecution(
+    selectedExecution?.workflow_id || "",
+    selectedExecution?.id || ""
+  );
+
+  useEffect(() => {
+    if (isOpen && selectedExecution) {
+      // You can add any side effects here when the sidebar opens
+    }
+  }, [isOpen, selectedExecution]);
+
+  if (!selectedExecution) return null;
+
+  const customFormatter = (value: number, unit: string, suffix: string) => {
+    if (!selectedExecution?.started) {
+      return "";
+    }
+
+    const formattedString = formatDistanceToNowStrict(
+      new Date(selectedExecution.started + "Z"),
+      { addSuffix: true }
+    );
+
+    return formattedString
+      .replace("about ", "")
+      .replace("minute", "min")
+      .replace("second", "sec")
+      .replace("hour", "hr");
+  };
+
+  return (
+    <Transition appear show={isOpen} as={Fragment}>
+      <Dialog onClose={toggle}>
+        <Transition.Child
+          as={Fragment}
+          enter="ease-out duration-300"
+          enterFrom="opacity-0"
+          enterTo="opacity-100"
+          leave="ease-in duration-200"
+          leaveFrom="opacity-100"
+          leaveTo="opacity-0"
+        >
+          <div className="fixed inset-0 bg-black/30 z-20" aria-hidden="true" />
+        </Transition.Child>
+        <Transition.Child
+          as={Fragment}
+          enter="transition ease-in-out duration-300 transform"
+          enterFrom="translate-x-full"
+          enterTo="translate-x-0"
+          leave="transition ease-in-out duration-300 transform"
+          leaveFrom="translate-x-0"
+          leaveTo="translate-x-full"
+        >
+          <Dialog.Panel className="fixed right-0 inset-y-0 w-2/4 bg-white z-30 p-6 overflow-auto flex flex-col">
+            <div className="flex justify-between mb-4">
+              <div>
+                <Dialog.Title className="text-3xl font-bold" as={Title}>
+                  Workflow Execution Details
+                  <Badge className="ml-4" color="orange">
```

Comment:
let's capitalize the status, which is now all in lowercase and pick color like icon, depending on the status value
```suggestion
                  <Badge className="ml-4 capitalize" color={selectedExecution.status === "error" ? "red" : selectedExecution.status === "success" ? "green" : "orange"}>
```